plugin/metrics/metrics.go

```diff
@@ -138,6 +145,10 @@ func keys(m map[string]bool) []string {
 // we listen on "localhost:0" and need to retrieve the actual address.
 var ListenAddr string
 
+// shutdownTimeout is the maximum amount of time the metrics plugin will wait
+// before erroring when it tries to close the metrics server
+var shutdownTimeout time.Duration = time.Second * 1
```

Comment:
> I do not like too much a global var.

Yeah, a constant would be better! I'll update the pull request accordingly.

> is 1 sec enough ?

I chose one second arbitrarily. Since this runs on reload, I figured that a shorter timeout would be better because it could (maybe?) suspend the entire reload process (i.e. a twenty second timeout might block the reload for twenty seconds).

I don't fully understand the reload plugin works and am making some assumptions. Would do you think?